### PR TITLE
Expand App status to include Enabled/Disabled and Running/Stopped distinctions

### PR DESCRIPTION
**Pull request by maackle**
TODO:

- [x] changelog
- [x] `todo!` sweep
- [x] ActiveAppsListed -> EnabledAppsListed

### CODE REVIEW

File: crates/holochain/src/conductor/api/api_external/admin_interface.rs [hunk 1]
```diff
@@ -214,23 +214,24 @@ impl AdminInterfaceApi for RealAdminInterfaceApi {
...
-            ListActiveApps => {
+            ListEnabledApps => {
                 tracing::warn!(
-                    "AdminRequest::ListActiveApps is deprecated, use AdminRequest::ListApps (TODO: update conductor-api)"
+                    "AdminRequest::ListEnabledApps is deprecated, use AdminRequest::ListApps (TODO: update conductor-api)"
                 );
 
-                let app_ids = self.conductor_handle.list_active_apps().await?;
+                let app_ids = self.conductor_handle.list_running_apps().await?;
                 Ok(AdminResponse::ActiveAppsListed(app_ids))
```
Comment (neonphog) [inline crates/holochain/src/conductor/api/api_external/admin_interface.rs:223]: should this be EnabledAppsListed?

File: crates/holochain/src/conductor/api/api_external/admin_interface.rs [hunk 2]
```diff
@@ -287,6 +297,22 @@ impl AdminInterfaceApi for RealAdminInterfaceApi {
...
+            ActivateApp { installed_app_id } => {
+                tracing::warn!("Admin method ActivateApp is deprecated: use EnableApp instead (functionality is identical).");
+                self.handle_admin_request_inner(EnableApp { installed_app_id })
+                    .await
+            }
+            DeactivateApp { installed_app_id } => {
+                tracing::warn!("Admin method DeactivateApp is deprecated: use DisableApp instead (functionality is identical).");
+                self.handle_admin_request_inner(DisableApp { installed_app_id })
+                    .await
+            }
```
Comment (neonphog) [inline crates/holochain/src/conductor/api/api_external/admin_interface.rs:315]: correct me if I'm wrong, but it looks to me like these deprecated calls are still returning the new results, so they wouldn't be compatible anyways? If that's true we should either not bother with the deprecation, or implement it fully with the result types as well.

File: crates/holochain/src/conductor/conductor.rs [hunk 3]
```diff
@@ -515,14 +515,30 @@ where
...
+        let stopped_cells: HashSet<CellId> = state
+            .stopped_apps()
+            .flat_map(|(_, app)| app.all_cells().cloned().collect::<HashSet<_>>())
+            .collect();
+        for (_, cell) in self
+            .cells
+            .iter()
+            .filter(|(id, _)| stopped_cells.contains(id))
+        {
+            cell.cell.cleanup().await?;
```
Comment (neonphog) [inline crates/holochain/src/conductor/conductor.rs:530]: is this cleanup likely to take time? perhaps we should return this for processing after the lock is released?

File: crates/holochain/src/conductor/manager/error.rs [hunk 4]
```diff
@@ -45,3 +45,17 @@ pub enum ManagedTaskError {
...
+
+impl ManagedTaskError {
+    pub fn is_recoverable(&self) -> bool {
+        use ConductorError as C;
+        use ManagedTaskError::*;
+        match self {
+            Io(_) | Join(_) | Recv(_) => false,
+            Conductor(err) => match err {
+                C::ShuttingDown => true,
+                _ => todo!("identify all recoverable cases"),
```
Comment (neonphog) [inline crates/holochain/src/conductor/manager/error.rs:57]: is this used yet? we may not want to merge this if it'll cause panics for unspecified error types

**Review by neonphog — CHANGES_REQUESTED** [review]
Couple things need looking at.

You can probably ignore the async cleanup holding open the lock comment... after looking through here, I see how everything else cell state related is doing that / understand why you were looking at cleaning that up.

I think it'll make sense to do in the not-to-distant future... it's no surprise when holo hammers a bunch of app installs all at once they take forever : )

File: crates/holochain/src/conductor/conductor.rs (hunk 3 above)
Comment (neonphog) [reply crates/holochain/src/conductor/conductor.rs:530]: can probably ignore this, given this is the state of the other calls in here : )

**Review by neonphog — APPROVED** [review]
@maackle - It still looks to me like there's a disconnect in the deprecated API...

I.e. on success, `ListEnabledApps` will give you an `EnabledAppsListed` struct, and the deprecated `ListActiveApps` will *also* give you an `EnabledAppsListed` struct, where previously it gave you an `ActiveAppsListed` struct... so won't this break clients using the deprecated api?

You're more familiar with the usage, so maybe this isn't a big deal. And otherwise looks good :+1:

**Comment by maackle** [comment]
@neonphog ah yes I meant to mention this -- I believe the struct name doesn't show up in the serialized response data, and the structure is the same for all of these deprecated structs and their counterparts, it should be OK.

**Pull request merged**